pkg/globalnet/controllers/ingress_endpoints_controller.go

```diff
@@ -0,0 +1,153 @@
+		ingressIPMap:         stringset.NewSynchronized(),
+	}
+
+	fieldSelector := fields.Set(map[string]string{"metadata.name": svc.Name}).AsSelector().String()
+
+	controller.resourceSyncer, err = syncer.NewResourceSyncer(&syncer.ResourceSyncerConfig{
+		Name:                "Ingress Endpoints syncer",
+		ResourceType:        &corev1.Endpoints{},
+		SourceClient:        config.SourceClient,
+		SourceNamespace:     svc.Namespace,
+		RestMapper:          config.RestMapper,
+		Federator:           federate.NewCreateFederator(config.SourceClient, config.RestMapper, corev1.NamespaceAll),
+		Scheme:              config.Scheme,
+		Transform:           controller.process,
+		SourceFieldSelector: fieldSelector,
+		ResourcesEquivalent: areEndpointsEqual,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err := controller.Start(); err != nil {
+		return nil, err
+	}
+
+	ingressIPs := config.SourceClient.Resource(*gvr).Namespace(corev1.NamespaceAll)
+	controller.reconcile(ingressIPs, func(obj *unstructured.Unstructured) runtime.Object {
+		// TODO: Consider adding endpointsRef field to GlobalIngressIPSpec
+		endpointsName, exists, _ := unstructured.NestedString(obj.Object, "spec", "podRef", "name")
+		if exists {
+			return &corev1.Endpoints{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      endpointsName,
+					Namespace: obj.GetNamespace(),
+				},
+			}
+		}
+
+		return nil
+	})
+
+	klog.Infof("Created Endpoints controller for (%s/%s) with selector %q", svc.Namespace, svc.Name, fieldSelector)
+
+	return controller, nil
+}
+
+func (c *ingressEndpointsController) process(from runtime.Object, numRequeues int, op syncer.Operation) (runtime.Object, bool) {
+	endpoints := from.(*corev1.Endpoints)
+	key, _ := cache.MetaNamespaceKeyFunc(endpoints)
+
+	ingressIP := &submarinerv1.GlobalIngressIP{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("ep-%.59s", endpoints.Name),
+			Namespace: endpoints.Namespace,
+			Labels: map[string]string{
+				ServiceRefLabel: c.svcName,
+			},
+		},
+	}
+
+	if op == syncer.Delete {
+		c.ingressIPMap.Remove(ingressIP.Name)
+		klog.Infof("ingress Endpoints %s for service %s deleted", key, c.svcName)
+
+		return ingressIP, false
+	}
+
+	// TODO: consider other conditions
+	if c.ingressIPMap.Contains(ingressIP.Name) {
+		// Avoid assigning ingressIPs to endpoints that are not ready with an endpoint IP
+		return nil, false
+	}
+
+	klog.Infof("%q ingress Endpoints %s for service %s", op, key, c.svcName)
+
+	// TODO: consider handling multiple endpoints IPs
+	if len(endpoints.Subsets) == 0 || len(endpoints.Subsets[0].Addresses) == 0 {
+		return nil, false
+	}
+
+	ingressIP.ObjectMeta.Annotations = map[string]string{
+		headlessSvcEndpointsIP: endpoints.Subsets[0].Addresses[0].IP,
+	}
+
+	ingressIP.Spec = submarinerv1.GlobalIngressIPSpec{
+		Target:     submarinerv1.HeadlessServiceEndpoints,
+		ServiceRef: &corev1.LocalObjectReference{Name: c.svcName},
+		// TODO: Consider adding endpointsRef field to GlobalIngressIPSpec
```

Comment:
I think we should definitely do this.

Reply:
Updated to add `endpointsRef`, however I started to think that it won't be necessary because name of Service and Endpoints are the same (I think that these fields are to track the full name for the case that it is truncated as a part of GlobalIngressIP's name, so ServiceRef field will be enough). 

If it is agreed, I will delete `endpointsRef` field.